File: backend/src/main/java/com/backend/petplace/domain/mypage/dto/MyPageUserInfo.java
package com.backend.petplace.domain.mypage.dto;

import com.backend.petplace.domain.user.entity.User;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class MyPageUserInfo {

  private final Long id;
  private final String nickname;
  private final String userEmail;
  private final LocalDateTime createdDate;
  private final String address;
  private final int point;
  private final int earnablePoints;
  private final int totalReviews;

  public static MyPageUserInfo from(User user, int earnablePoints, int totalReviews){ //dto 내부 조립이 필요하면 from을 씁시다
    return MyPageUserInfo.builder()
        .id(user.getId())
        .nickname(user.getNickName())
        .userEmail(user.getEmail())
        .createdDate(user.getCreatedDate())
        .address(user.getAddress() + " " + user.getAddressDetail())
        .point(user.getTotalPoint())
        .earnablePoints(earnablePoints)
        .totalReviews(totalReviews)
        .build();
  }
}

File: backend/src/main/java/com/backend/petplace/domain/mypage/dto/MyPageUserInfo.kt
package com.backend.petplace.domain.mypage.dto

import com.backend.petplace.domain.user.entity.User
import java.time.LocalDateTime

data class MyPageUserInfo(
    val id: Long,
    val nickname: String,
    val userEmail: String,
    val createdDate: LocalDateTime?, //createdDate가 nullable이므로
    val address: String,
    val point: Int,
    val earnablePoints: Int,
    val totalReviews: Int
) {
    companion object {
        fun from(user: User, earnablePoints: Int, totalReviews: Int): MyPageUserInfo {
            return MyPageUserInfo(
                id = user.id!!,
                nickname = user.nickName,
                userEmail = user.email,
                createdDate = user.createdDate,
                address = "${user.address} ${user.addressDetail}",
                point = user.totalPoint,
                earnablePoints = earnablePoints,
                totalReviews = totalReviews
            )
        }
    }
}